Examples/Movies/SearchScreen.js

```diff
@@ -215,37 +215,38 @@ var SearchScreen = React.createClass({
         });
       })
       .done();
-  },
+  };
 
-  getDataSource: function(movies: Array<any>): ListView.DataSource {
+  getDataSource(movies: Array<any>): ListView.DataSource {
     return this.state.dataSource.cloneWithRows(movies);
-  },
+  }
 
-  selectMovie: function(movie: Object) {
+  selectMovie(movie: Object) {
+    const { navigator } = this.props;
     if (Platform.OS === 'ios') {
-      this.props.navigator.push({
+      navigator.push({
         title: movie.title,
         component: MovieScreen,
         passProps: {movie},
       });
     } else {
       dismissKeyboard();
-      this.props.navigator.push({
+      navigator.push({
         title: movie.title,
         name: 'movie',
         movie: movie,
       });
     }
-  },
+  }
 
-  onSearchChange: function(event: Object) {
+  onSearchChange = (event: Object) => {
     var filter = event.nativeEvent.text.toLowerCase();
 
-    this.clearTimeout(this.timeoutID);
-    this.timeoutID = this.setTimeout(() => this.searchMovies(filter), 100);
-  },
+    clearTimeout(this.timeoutID);
+    this.timeoutID = setTimeout(() => this.searchMovies(filter), 100);
+  };
 
-  renderFooter: function() {
+  renderFooter = () => {
```

Comment:
Well, how to avoid this? I can refactor these class methods, and then use bind or arrow function(which do you prefer?), but how to handle `timeoutID`?